src/ui/handler_manager.js

```diff
@@ -501,13 +446,6 @@ class HandlerManager {
             around = pinchAround;
         }
 
-        if (eventStarted("drag") && around) {
-            this._dragOrigin = toVec3(tr.pointCoordinate3D(around));
-            // Construct the tracking ellipsoid every time user changes the drag origin.
-            // Direction of the ray will define size of the shape and hence defining the available range of movement
-            this._trackingEllipsoid.setup(tr._camera.position, this._dragOrigin);
```

Comment:
This was a good call to remove this, but investigating that further, I think that the tracking ellipsoid ended up being unused incorrectly as a result of introducing globe (only realized that by testing this PR), it's a subtle difference that has gone unnoticed, but the original effect should be that drag pan is compensated and smoother on pitched views. So I think we should try to re-introduce it as it's broken in `main`, to help drag pan on pitched views.

Looking at the native code, there is a switch that happens when the projection is mercator [here](https://github.com/mapbox/mapbox-gl-native-internal/blob/8a09d55875059f287019a5fe69e06dafd37936a2/src/mbgl/map/transform.cpp#L559) and we should at least have code that projects a ray (the part that is missing for it to fully work), something like:
```js
...
if (tr.projection.name === 'mercator') {
    assert(this._dragOrigin, '_dragOrigin should have been setup with a previous dragstart');
    const startPoint = this._trackingEllipsoid.projectRay(tr.screenPointToMercatorRay(around).dir);
    const endPoint = this._trackingEllipsoid.projectRay(tr.screenPointToMercatorRay(around.sub(panDelta)).dir);
    panVec[0] = endPoint[0] - startPoint[0];
    panVec[1] = endPoint[1] - startPoint[1];
}
...
```

<details><summary>Here's a full diff  that I quickly tested locally:</summary>

```diff
diff --git a/src/ui/handler_manager.js b/src/ui/handler_manager.js
index 8f8c7d542..4e6dcc2a1 100644
--- a/src/ui/handler_manager.js
+++ b/src/ui/handler_manager.js
@@ -36,6 +36,50 @@ class RenderFrameEvent extends Event {
     timeStamp: number;
 }
 
+class TrackingEllipsoid {
+    constants: Array<number>;
+    radius: number;
+
+    constructor() {
+        // a, b, c in the equation x²/a² + y²/b² + z²/c² = 1
+        this.constants = [1, 1, 0.01];
+        this.radius = 0;
+    }
+
+    setup(center: Vec3, pointOnSurface: Vec3) {
+        const centerToSurface = vec3.sub([], pointOnSurface, center);
+        if (centerToSurface[2] < 0) {
+            this.radius = vec3.length(vec3.div([], centerToSurface, this.constants));
+        } else {
+            // The point on surface is above the center. This can happen for example when the camera is
+            // below the clicked point (like a mountain) Use slightly shorter radius for less aggressive movement
+            this.radius = vec3.length([centerToSurface[0], centerToSurface[1], 0]);
+        }
+    }
+
+    // Cast a ray from the center of the ellipsoid and the intersection point.
+    projectRay(dir: Vec3): Vec3 {
+        // Perform the intersection test against a unit sphere
+        vec3.div(dir, dir, this.constants);
+        vec3.normalize(dir, dir);
+        vec3.mul(dir, dir, this.constants);
+
+        const intersection = vec3.scale([], dir, this.radius);
+
+        if (intersection[2] > 0) {
+            // The intersection point is above horizon so special handling is required.
+            // Otherwise direction of the movement would be inverted due to the ellipsoid shape
+            const h = vec3.scale([], [0, 0, 1], vec3.dot(intersection, [0, 0, 1]));
+            const r = vec3.scale([], vec3.normalize([], [intersection[0], intersection[1], 0]), this.radius);
+            const p = vec3.add([], intersection, vec3.scale([], vec3.sub([], vec3.add([], r, h), intersection), 2));
+
+            intersection[0] = p[0];
+            intersection[1] = p[1];
+        }
+
+        return intersection;
+    }
+}
 // Handlers interpret dom events and return camera changes that should be
 // applied to the map (`HandlerResult`s). The camera changes are all deltas.
 // The handler itself should have no knowledge of the map's current state.
@@ -111,6 +155,8 @@ class HandlerManager {
     _changes: Array<[HandlerResult, Object, any]>;
     _previousActiveHandlers: { [string]: Handler };
     _listeners: Array<[HTMLElement, string, void | EventListenerOptionsOrUseCapture]>;
+    _trackingEllipsoid: TrackingEllipsoid;
+    _dragOrigin: ?Vec3;
 
     constructor(map: Map, options: { interactive: boolean, pitchWithRotate: boolean, clickTolerance: number, bearingSnap: number}) {
         this._map = map;
@@ -122,6 +168,8 @@ class HandlerManager {
         this._inertia = new HandlerInertia(map);
         this._bearingSnap = options.bearingSnap;
         this._previousActiveHandlers = {};
+        this._trackingEllipsoid = new TrackingEllipsoid();
+        this._dragOrigin = null;
 
         // Track whether map is currently moving, to compute start/move/end events
         this._eventsInProgress = {};
@@ -416,18 +464,24 @@ class HandlerManager {
         this._changes = [];
     }
 
-    _eventEnded(type: string): boolean {
-        const event = this._eventsInProgress[type];
-        return event && !this._handlersById[event.handlerName].isActive();
-    }
-
     _updateMapTransform(combinedResult: any, combinedEventsInProgress: Object, deactivatedHandlers: Object) {
 
         const map = this._map;
         const tr = map.transform;
+
+        const eventStarted = (type) => {
+            const newEvent = combinedEventsInProgress[type];
+            return newEvent && !this._eventsInProgress[type];
+        };
+
+        const eventEnded = (type) => {
+            const event = this._eventsInProgress[type];
+            return event && !this._handlersById[event.handlerName].isActive();
+        };
+
         const toVec3 = (p: MercatorCoordinate): Vec3 => [p.x, p.y, p.z];
 
-        if (this._eventEnded("drag") && !hasChange(combinedResult)) {
+        if (eventEnded("drag") && !hasChange(combinedResult)) {
             const preZoom = tr.zoom;
             tr.cameraElevationReference = "sea";
             tr.recenterOnTerrain();
@@ -446,6 +500,13 @@ class HandlerManager {
             around = pinchAround;
         }
 
+        if (eventStarted("drag") && around) {
+            this._dragOrigin = toVec3(tr.pointCoordinate3D(around));
+            // Construct the tracking ellipsoid every time user changes the drag origin.
+            // Direction of the ray will define size of the shape and hence defining the available range of movement
+            this._trackingEllipsoid.setup(tr._camera.position, this._dragOrigin);
+        }
+
         // All movement of the camera is done relative to the sea level
         tr.cameraElevationReference = "sea";
 
@@ -460,22 +521,30 @@ class HandlerManager {
         // Compute Mercator 3D camera offset based on screenspace panDelta
         const panVec = [0, 0, 0];
         if (panDelta) {
-            const startPoint = tr.pointCoordinate(around);
-            if (tr.projection.name === 'globe') {
-                // Compute pan vector directly in pixel coordinates for the globe.
-                // Rotate the globe a bit faster when dragging near poles to compensate
-                // different pixel-per-meter ratios (ie. pixel-to-physical-rotation is lower)
-                panDelta = panDelta.rotate(-tr.angle);
-                const scale = tr._pixelsPerMercatorPixel / tr.worldSize;
-                panVec[0] = -panDelta.x * mercatorScale(latFromMercatorY(startPoint.y)) * scale;
-                panVec[1] = -panDelta.y * mercatorScale(tr.center.lat) * scale;
-
+            if (tr.projection.name === 'mercator') {
+                assert(this._dragOrigin, '_dragOrigin should have been setup with a previous dragstart');
+                const startPoint = this._trackingEllipsoid.projectRay(tr.screenPointToMercatorRay(around).dir);
+                const endPoint = this._trackingEllipsoid.projectRay(tr.screenPointToMercatorRay(around.sub(panDelta)).dir);
+                panVec[0] = endPoint[0] - startPoint[0];
+                panVec[1] = endPoint[1] - startPoint[1];
             } else {
-                const endPoint = tr.pointCoordinate(around.sub(panDelta));
-
-                if (startPoint && endPoint) {
-                    panVec[0] = endPoint.x - startPoint.x;
-                    panVec[1] = endPoint.y - startPoint.y;
+                const startPoint = tr.pointCoordinate(around);
+                if (tr.projection.name === 'globe') {
+                    // Compute pan vector directly in pixel coordinates for the globe.
+                    // Rotate the globe a bit faster when dragging near poles to compensate
+                    // different pixel-per-meter ratios (ie. pixel-to-physical-rotation is lower)
+                    panDelta = panDelta.rotate(-tr.angle);
+                    const scale = tr._pixelsPerMercatorPixel / tr.worldSize;
+                    panVec[0] = -panDelta.x * mercatorScale(latFromMercatorY(startPoint.y)) * scale;
+                    panVec[1] = -panDelta.y * mercatorScale(tr.center.lat) * scale;
+
+                } else {
+                    const endPoint = tr.pointCoordinate(around.sub(panDelta));
+
+                    if (startPoint && endPoint) {
+                        panVec[0] = endPoint.x - startPoint.x;
+                        panVec[1] = endPoint.y - startPoint.y;
+                    }
                 }
             }
         }

```

</details>